src/core/net/dns_headers.hpp

```diff
@@ -382,6 +618,54 @@ OT_TOOL_PACKED_BEGIN
 class ResourceRecord
 {
 public:
+    /**
+     * Resource Record Types.
+     *
+     */
+    enum : uint16_t
+    {
+        kTypeA    = 1,  ///< Address record (IPv4).
+        kTypePtr  = 12, ///< PTR record.
+        kTypeTxt  = 16, ///< TXT record.
+        kTypeSrv  = 33, ///< Service locator record.
```

Comment:
nit: I don't think "service locator" is the right term. What about being consistent with the ones above?
```suggestion
        kTypeSrv  = 33, ///< SRV locator record.
```